packages/tofu/src/inspectSource.js

```diff
@@ -175,60 +172,55 @@ function inspectGlobals(
 }
 
 /**
+ * This finds all modules `import`ed into the AST, as well as any re-exported
+ * modules.
  *
- * @param {import('../../core/src/generatePolicy').AST} ast
- * @returns {{esmImports: string[]}}
+ * @param {import('@babel/types').Node} ast
+ * @param {string[]} [packagesToInspect] - List of module IDs to look for; if
+ *   omitted, will inspect all imports
+ * @returns {string[]} - List of imported identifiers
  */
-function inspectEsmImports(ast) {
-  /** @type {string[]} */
-  const esmImports = []
+function inspectEsmImports(ast, packagesToInspect) {
+  const pkgsToInspect = new Set(packagesToInspect)
+
+  /** @type {Set<string>} */
+  const esmImports = new Set()
+
+  /**
+   * @param {import('@babel/traverse').NodePath<
+   *   | import('@babel/types').ImportDeclaration
+   *   | import('@babel/types').ExportNamedDeclaration
+   * >} path
+   */
+  const handleNodePath = (path) => {
+    const importSource = path.node.source?.value
+
+    if (
+      importSource &&
+      ((packagesToInspect && pkgsToInspect.has(importSource)) ||
+        !packagesToInspect)
+    ) {
+      esmImports.add(importSource)
+    }
+  }
+
   traverse(ast, {
-    ImportDeclaration: (path) => {
-      const { node } = path
-      const { specifiers, source } = node
-      // not sure if this is ever not a StringLiteral, but just in case
-      if (source.type !== 'StringLiteral') {
-        return
-      }
-      const importSource = source.value
-      specifiers.forEach((spec) => {
```

Comment:
How is all this no longer needed? I'm gonna need to dig into this some more. For now what I'm seeing here is we add stuff to imports regardless of all the specifiers processing that was happening here and the case where importName is modified is no longer handled. Not sure why.

Reply:
`ImportDeclaration` handles everything we care about

Reply:
You sound very certain. I have no reason to disbelieve that until we see stuff missing from policies.